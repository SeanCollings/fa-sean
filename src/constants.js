export const API_THE_ICONIC_URL =
  'https://eve.theiconic.com.au/catalog/products';

export const FILTER_SORT = {
  popularity: { display: 'Popularity', field: 'popularity' },
  new: { display: 'New Arrivals', field: 'new' },
  price_high: { display: 'Price High to Low', field: 'price_high' },
  price_low: { display: 'Price Low to High', field: 'price_low' },
  brand_asc: { display: 'Brand A to Z', field: 'brand_asc' },
  brand_desc: { display: 'Brand Z to A', field: 'brand_desc' },
};

export const FILTER_GENDER = {
  female: { display: 'Women', field: 'female' },
  male: { display: 'Men', field: 'male' },
  kids: { display: 'Kids', field: 'kids' },
};

export const COLOURS = {
  dark: '#232323',
  gray: '#eee',
  light: '#f7f7f7;',
  textDark: '#4d4d4d',
  textGray: '#909090',
  red: '#ef423c',
  white: '#ffffff',
  errorText: '#ee110d',
  errorBackground: '#fcecec',
};

// Â»
