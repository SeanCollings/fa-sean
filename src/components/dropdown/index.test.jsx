import React from 'react';
import renderer from 'react-test-renderer';
import { FILTER_SORT } from '../../constants';
import Dropdown from '.';

describe('dropdown unit test', () => {
  const id = 'dropdown_test';
  const onChange = jest.fn();
  const items = Object.values(FILTER_SORT);

  test('should render without label', () => {
    const component = renderer.create(
      <Dropdown name={id} id={id} onChange={onChange} items={items} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render with label', () => {
    const component = renderer.create(
      <Dropdown
        name={id}
        id={id}
        onChange={onChange}
        items={items}
        labelText="labelText_test"
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
