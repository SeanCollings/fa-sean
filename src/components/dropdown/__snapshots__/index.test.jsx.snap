// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dropdown unit test should render with label 1`] = `
Array [
  <label
    className="sc-bdfBwQ iEvZEX"
    htmlFor="dropdown_test"
  >
    labelText_test
  </label>,
  <select
    className="sc-gsTCUz dmkZhF"
    id="dropdown_test"
    name="dropdown_test"
    onChange={[MockFunction]}
  >
    <option
      value="popularity"
    >
      Popularity
    </option>
    <option
      value="new"
    >
      New Arrivals
    </option>
    <option
      value="price_high"
    >
      Price High to Low
    </option>
    <option
      value="price_low"
    >
      Price Low to High
    </option>
    <option
      value="brand_asc"
    >
      Brand A to Z
    </option>
    <option
      value="brand_desc"
    >
      Brand Z to A
    </option>
  </select>,
]
`;

exports[`dropdown unit test should render without label 1`] = `
<select
  className="sc-gsTCUz dmkZhF"
  id="dropdown_test"
  name="dropdown_test"
  onChange={[MockFunction]}
>
  <option
    value="popularity"
  >
    Popularity
  </option>
  <option
    value="new"
  >
    New Arrivals
  </option>
  <option
    value="price_high"
  >
    Price High to Low
  </option>
  <option
    value="price_low"
  >
    Price Low to High
  </option>
  <option
    value="brand_asc"
  >
    Brand A to Z
  </option>
  <option
    value="brand_desc"
  >
    Brand Z to A
  </option>
</select>
`;
